actions/approve_csr_handler.go

```diff
@@ -122,25 +131,28 @@ func (h *approveCSRHandler) getInitialNodeCSR(ctx context.Context, log logrus.Fi
 	var cert *csr.Certificate
 	var err error
 
-	logRetry := func(err error, _ time.Duration) {
-		log.Warnf("getting initial csr, will retry: %v", err)
-	}
-	b := backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3)
-	err = backoff.RetryNotify(func() error {
-		cert, err = poll()
-		if errors.Is(err, context.DeadlineExceeded) {
-			return backoff.Permanent(err)
-		}
-		return err
-	}, b, logRetry)
+	b := waitext.DefaultExponentialBackoff()
+	err = waitext.Retry(
+		ctx,
+		b,
+		3,
+		func(ctx context.Context) (bool, error) {
+			cert, err = poll()
+			if errors.Is(err, context.DeadlineExceeded) {
+				return false, err
+			}
+			return true, err
+		},
+		func(err error) {
+			log.Warnf("getting initial csr, will retry: %v", err)
+		},
+	)
 
 	return cert, err
 }
 
-func newApproveCSRExponentialBackoff() *backoff.ExponentialBackOff {
-	b := backoff.NewExponentialBackOff()
-	b.Multiplier = 2
-	b.MaxElapsedTime = 4 * time.Minute
-	b.Reset()
+func newApproveCSRExponentialBackoff() wait.Backoff {
+	b := waitext.DefaultExponentialBackoff()
+	b.Factor = 2
```

Comment:
Not to run this forever and preserve old `b.MaxElapsedTime` behaviour.
```suggestion
	b.Factor = 2
	b..Cap = 4 * time.Minute
```